app/scripts/components/boilerplate/language-selector.jsx

```diff
@@ -1,14 +1,28 @@
 import React from 'react';
+import classNames from 'classnames';
 import T from '../utils/t';
 
 require('stylesheets/boilerplate/language-selector');
 
 const LanguageSelector = React.createClass({
+  getInitialState() {
+    return {lang: 'en', langList: ['en', 'sw-tz']};
+  },
+
+  handleClick() {
+    // TODO to be implemented
+  },
+
   render() {
     return (
       <ul className="language-selector">
-        <li><T k="lang.en" /></li>
-        <li><T k="lang.sw-tz" /></li>
+        {this.state.langList.map(function(item, index) {
+          const classes = classNames({
+            'language-selector': true,
+            'active': this.state.lang === item,
+          });
+          return <li className={classes}  key={index} onClick={this.handleClick.bind(this, item)}><T k={`lang.${item}`} /></li>;
```

Comment:
actually there is a jsx syntax `::` for binding to a component callback, which I've seen used in react-leaflet: https://github.com/PaulLeCam/react-leaflet/blob/master/example/events.js#L40 might be worth trying?